skimage/feature/tests/test_corner.py

```diff
@@ -41,30 +41,37 @@ def test_structure_tensor():
 def test_hessian_matrix():
     square = np.zeros((5, 5))
     square[2, 2] = 4
-    Hxx, Hxy, Hyy = hessian_matrix(square, sigma=0.1)
-    assert_almost_equal(Hxx, np.array([[0, 0,  0, 0, 0],
+    Hrr, Hrc, Hcc = hessian_matrix(square, sigma=0.1, order='rc')
+    assert_almost_equal(Hrr, np.array([[0, 0,  0, 0, 0],
                                        [0, 0,  0, 0, 0],
                                        [2, 0, -2, 0, 2],
                                        [0, 0,  0, 0, 0],
                                        [0, 0,  0, 0, 0]]))
 
-    assert_almost_equal(Hxy, np.array([[0,  0, 0,  0, 0],
+    assert_almost_equal(Hrc, np.array([[0,  0, 0,  0, 0],
                                        [0,  1, 0, -1, 0],
                                        [0,  0, 0,  0, 0],
                                        [0, -1, 0,  1, 0],
                                        [0,  0, 0,  0, 0]]))
 
-    assert_almost_equal(Hyy, np.array([[0, 0,  2, 0, 0],
+    assert_almost_equal(Hcc, np.array([[0, 0,  2, 0, 0],
                                        [0, 0,  0, 0, 0],
                                        [0, 0, -2, 0, 0],
                                        [0, 0,  0, 0, 0],
                                        [0, 0,  2, 0, 0]]))
 
+    matrix2d = np.random.rand(3,3)
+    Arr, Arc, Acc = hessian_matrix(matrix2d, sigma=0.1)
```

Comment:
I suspect that this call is what's causing Python 2 not to warn — the warning might only be raised once in Py2 for some reason. Since this line is doing nothing useful, just remove it — the `assert_warns` function will call `hessian_matrix` as appropriate.